app/assets/javascripts/components/story/StoryAttachment.js

```diff
@@ -0,0 +1,91 @@
+import React from 'react';
+
+class StoryAttachment extends React.Component {
+  constructor(props) {
+    super(props);
+    this.saveInput = (input) => { this.filesInput = input };
+    this.state = {
+      random: (Math.floor(Math.random() * 10000) + 1),
+      progressElementId: '',
+      finishedElementId: '',
+      attachinaryContainerId: ''
+    }
+  }
+
+  componentWillMount() {
+    this.setState({
+      progressElementId: "documents_progress_" + this.state.random,
+      finishedElementId: "documents_finished_" + this.state.random,
+      attachinaryContainerId: "attachinary_container_" + this.state.random
+    });
+  }
+
+  AttachmentInput() {
+    const { name, attachinaryProps } = this.props;
+    let attachinary = attachinaryProps(
+      'documents',
+      this.state.progressElementId,
+      this.state.finishedElementId,
+      this.state.attachinaryContainerId
+    );
+
+    return(
+      <input
+        type='file'
+        name={name}
+        className='attachinary-input'
+        ref={this.saveInput}
+        multiple={attachinary.multiple}
+        data-attachinary={attachinary.dataAttachinary}
+        data-form-data={attachinary.dataFormData}
+        data-url={attachinary.dataUrl}
+      />
+    );
+  }
+
+  progressBar() {
+    return(
+      <div
+        id={this.state.progressElementId}
+        className='attachinary_progress_bar'
+      />
+    );
+  }
+
+  render() {
+    const { name, isReadonly } = this.props;
+    return(
+      <div className="uploads">
+        <label htmlFor={name}>{ I18n.t(`story.${name}`) }</label>
+        { (!isReadonly) ? this.AttachmentInput() : null }
+        { (!isReadonly) ? this.progressBar() : null }
+        <div id={this.state.attachinaryContainerId} />
+      </div>
+    );
+  }
+
+  componentDidMount() {
```

Comment:
Prefer to have lifecycle React methods as the first ones, right after the constructor.